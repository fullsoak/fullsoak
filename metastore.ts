const store = new Map();

const FULLSOAK_APP_COMPONENTS_DIR = "FULLSOAK_APP_COMPONENTS_DIR";

const setGlobalComponentsDir = (dir: string) =>
  store.set(FULLSOAK_APP_COMPONENTS_DIR, dir);
const getGlobalComponentsDir = store.get(FULLSOAK_APP_COMPONENTS_DIR);

export { getGlobalComponentsDir, setGlobalComponentsDir };
