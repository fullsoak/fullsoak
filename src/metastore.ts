import type { ApplicationListenEvent } from "@oak/oak/application";
import { LogDebug } from "./utils.ts";

const store = new Map();

const FULLSOAK_APP_COMPONENTS_DIR = "FULLSOAK_APP_COMPONENTS_DIR";
const APP_LISTEN_OBJ = "APP_LISTEN_OBJ";
export const setGlobalComponentsDir = (dir: string) => {
  store.set(FULLSOAK_APP_COMPONENTS_DIR, dir);
  LogDebug("set global components dir", dir);
};

export const getGlobalComponentsDir = (): string =>
  store.get(FULLSOAK_APP_COMPONENTS_DIR);

export const setAppListenObj = (obj: ApplicationListenEvent) =>
  store.set(APP_LISTEN_OBJ, obj);

const getAppListenObj = (): ApplicationListenEvent => store.get(APP_LISTEN_OBJ);

/**
 * server-side only
 */
export const getOrigin = (): string => {
  const obj = getAppListenObj();
  const scheme = obj.secure ? "https" : "http";
  const localOrigin = obj.port != null
    ? `${scheme}://${obj.hostname}:${obj.port}`
    : obj.hostname;
  return localOrigin;
};
