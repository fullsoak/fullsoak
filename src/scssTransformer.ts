import { IS_DEBUG } from "./getEnv.ts";

let sassEmbedded: typeof import("sass-embedded");

const getScssTransformer = async () => {
  return sassEmbedded || (sassEmbedded = await import("sass-embedded"));
};

export const transformScss = async (path: string): Promise<string> => {
  try {
    const res = await (await getScssTransformer()).compileAsync(path);
    // @NOTE there are other fields under `res`
    return res.css;
  } catch (e) {
    if (IS_DEBUG) console.error(e);
    const err = e as { sassMessage: string; stack: string; message: string };
    const normalizedErr = new Error(err.message || err.sassMessage);
    normalizedErr.stack = err.stack;
    throw normalizedErr;
  }
};
