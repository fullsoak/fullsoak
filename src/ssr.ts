import type { FunctionComponent, VNode } from "preact";
import type { CP } from "./types.ts";
import { renderToStringAsync } from "preact-render-to-string";
import { withHtmlShell } from "./HtmlShell.tsx";
import { cleanCss } from "./minifyCss.ts";
import { html } from "htm/preact";
import { getComponentCss } from "./getComponentCss.ts";

/**
 * takes a component that starts from the root `<html>` element
 * and renders it out as a string; the input component is usually
 * constructed by using the {@link HtmlShell} tsx component (included battery)
 *
 * @example
 * ```tsx
 * import { byoHtml, HtmlShell } from "@fullsoak/fullsoak";
 * byoHtml(
 *   <HtmlShell
 *     componentName="MyApp"
 *     js="some vanilla javascript"
 *     css="some raw css"
 *   />
 * );
 * ```
 */
export const byoHtml = async (component: VNode): Promise<string> => {
  globalThis.location = {} as Location;
  return "<!doctype html>" + (await renderToStringAsync(component));
};

const ssrVNode = async (
  node: VNode,
): Promise<string> => {
  const componentName = typeof node.type === "string"
    ? node.type
    : node.type.name;
  const componentCss = await getComponentCss(componentName);
  return byoHtml(withHtmlShell({
    componentName,
    componentProps: node.props,
    component: node,
    css: cleanCss(componentCss),
  }));
};

const ssrTsxComponent = async <P extends CP>(
  component: FunctionComponent<P>,
  props: P | null = null,
): Promise<string> => {
  const componentName = component.name;

  // this is the only way to render a TSX component where the
  // deepest DOM element is rendered (which is the whole point of SSR)
  const componentVNode = html`<${component} ...${props} />`;

  const componentCss = await getComponentCss(componentName);
  return byoHtml(withHtmlShell({
    componentName,
    componentProps: props,
    // while using `h` also works, it doesn't render the deepest DOM element
    // component: h<CP>(componentName, props),
    component: componentVNode,
    css: cleanCss(componentCss),
  }));
};

/**
 * takes a Preact `VNode` or a TSX component and wraps it around an `<html>` element
 * then renders everything out as a string
 * @param renderTarget - a Preact VNode or a TSX component
 * @param componentProps - the props to pass to the component (only applicable when passing a TSX component as `renderTarget`)
 * @returns - the rendered HTML as a string
 * @TODO update the README example
 */
export const ssr = <P extends CP>(
  renderTarget: FunctionComponent<P> | VNode,
  componentProps: P | null = null,
): Promise<string> => {
  // @TODO see if there's a better way to check if it's a VNode
  const isVNode = typeof renderTarget === "object" && "type" in renderTarget;
  if (isVNode) return ssrVNode(renderTarget);
  return ssrTsxComponent(renderTarget, componentProps);
};
