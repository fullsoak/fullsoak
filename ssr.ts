import type { FunctionComponent, VNode } from "preact";
import { render } from "preact-render-to-string";
import { withHtmlShell } from "./HtmlShell.tsx";
import { cleanCss } from "./minifyCss.ts";
import { html } from "htm/preact";
import { getGlobalComponentsDir } from "./metastore.ts";

/**
 * takes a component that starts from the 'root' <html> element
 * and render it out as a string
 */
export const byoHtml = (component: VNode): string =>
  "<!doctype html>" + render(component);

/**
 * takes a standard Preact component and wrap it around an <html> element
 * and render it out as a string
 */
export const ssr = async (component: FunctionComponent): Promise<string> => {
  const componentName = component.name;
  const componentVNode = html`<${component} />`;
  let componentCss;
  try {
    // @TODO use a framework smart fn that attempts to read all .css files in the `componentName` dir?
    // @TODO also consider the possibility to combine a general 'main.css' and a component-specific css
    componentCss = await Deno.readTextFile(
      `${getGlobalComponentsDir()}/${componentName}/styles.css`,
    );
  } catch (e) {
    console.warn(
      `unable to load css for component ${componentName}: ${
        (e as Error).message
      }`,
    );
  }
  return byoHtml(withHtmlShell({
    component: componentVNode,
    css: cleanCss(componentCss),
  }));
};
