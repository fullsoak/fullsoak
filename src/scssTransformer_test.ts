import { join } from "@std/path";
import { transformScss } from "./scssTransformer.ts";
import { assertSnapshot } from "@std/testing/snapshot";
import { ssr } from "./ssr.ts";
import { StandardComponent } from "@mocks/components/StandardComponent/index.tsx";
import { setGlobalComponentsDir } from "./metastore.ts";

// this is called in `useFullsoak`, but we call it here to make unit tests simpler
setGlobalComponentsDir("./mocks/components");

Deno.test("transformScss", async (t) => {
  const out = await transformScss(
    join(
      import.meta.dirname || "",
      "../mocks/components/StandardComponent/styles.scss",
    ),
  );
  await assertSnapshot(t, out);
});

Deno.test("StandardComponent", async (t) => {
  const output = await ssr(StandardComponent);
  // console.log(output);
  await assertSnapshot(t, output);
});
