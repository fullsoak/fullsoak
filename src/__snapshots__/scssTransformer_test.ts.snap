export const snapshot = {};

snapshot[`transformScss 1`] = `
'body {
  background-color: #fbfbfb;
  background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDAgMTAwIj4KICA8IS0tIEJhc2UgdHJhbnNwYXJlbnQgYmFja2dyb3VuZCAtLT4KICA8cmVjdCB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgZmlsbD0idHJhbnNwYXJlbnQiLz4KICAKICA8IS0tIFN1YnRsZSBnZW9tZXRyaWMgcGF0dGVybiAtLT4KICA8ZyBmaWxsPSIjMzMzMzMzIiBmaWxsLW9wYWNpdHk9IjAuMDMiPgogICAgPCEtLSBTbWFsbCBjaXJjbGVzIC0tPgogICAgPGNpcmNsZSBjeD0iMTAiIGN5PSIxMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSIzMCIgY3k9IjEwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjUwIiBjeT0iMTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNzAiIGN5PSIxMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI5MCIgY3k9IjEwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMjAiIGN5PSIyMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI0MCIgY3k9IjIwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjYwIiBjeT0iMjAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iODAiIGN5PSIyMCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjEwIiBjeT0iMzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iMzAiIGN5PSIzMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI1MCIgY3k9IjMwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjcwIiBjeT0iMzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iOTAiIGN5PSIzMCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjIwIiBjeT0iNDAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNDAiIGN5PSI0MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI2MCIgY3k9IjQwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjgwIiBjeT0iNDAiIHI9IjEuNSIvPgogICAgCiAgICA8Y2lyY2xlIGN4PSIxMCIgY3k9IjUwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjMwIiBjeT0iNTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNTAiIGN5PSI1MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI3MCIgY3k9IjUwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjkwIiBjeT0iNTAiIHI9IjEuNSIvPgogICAgCiAgICA8Y2lyY2xlIGN4PSIyMCIgY3k9IjYwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjQwIiBjeT0iNjAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNjAiIGN5PSI2MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI4MCIgY3k9IjYwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMTAiIGN5PSI3MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSIzMCIgY3k9IjcwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjUwIiBjeT0iNzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNzAiIGN5PSI3MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI5MCIgY3k9IjcwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMjAiIGN5PSI4MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI0MCIgY3k9IjgwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjYwIiBjeT0iODAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iODAiIGN5PSI4MCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjEwIiBjeT0iOTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iMzAiIGN5PSI5MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI1MCIgY3k9IjkwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjcwIiBjeT0iOTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iOTAiIGN5PSI5MCIgcj0iMS41Ii8+CiAgICAKICAgIDwhLS0gVGhpbiBsaW5lcyAtLT4KICAgIDxsaW5lIHgxPSIwIiB5MT0iMjUiIHgyPSIxMDAiIHkyPSIyNSIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSIwIiB5MT0iNzUiIHgyPSIxMDAiIHkyPSI3NSIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSIyNSIgeTE9IjAiIHgyPSIyNSIgeTI9IjEwMCIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSI3NSIgeTE9IjAiIHgyPSI3NSIgeTI9IjEwMCIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICA8L2c+Cjwvc3ZnPg==");
  background-repeat: repeat;
  background-size: 200px 200px;
  background-attachment: fixed;
}'
`;

snapshot[`StandardComponent 1`] = `'<!doctype html><html><head><meta charset="utf-8"/><meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0"/><script type="importmap">{"imports":{"preact":"https://esm.sh/preact@10.26.5","preact/hooks":"https://esm.sh/preact@10.26.5/hooks","preact/debug":"https://esm.sh/preact@10.26.5/debug","preact/devtools":"https://esm.sh/preact@10.26.5/devtools","react":"https://esm.sh/preact@10.26.5","react/jsx-runtime":"https://esm.sh/preact@10.26.5/jsx-runtime","preact/jsx-runtime":"https://esm.sh/preact@10.26.5/jsx-runtime","react-dom":"https://esm.sh/preact@10.26.5/compat/","react-dom/*":"https://esm.sh/preact@10.26.5/compat/*","htm/preact":"https://esm.sh/htm@3.1.1/preact?external=preact","preact-iso":"https://esm.sh/preact-iso@^2.9.1?external=preact","preact-iso/prerender":"/fullsoak","@preact/signals":"https://esm.sh/@preact/signals","fullsoak/preact-iso":"https://esm.sh/preact-iso@^2.9.1?external=preact","@fullsoak/fullsoak/preact-iso":"https://esm.sh/preact-iso@^2.9.1?external=preact","fullsoak":"/fullsoak","@fullsoak/fullsoak":"/fullsoak"}}</script><link rel="modulepreload" href="/components/StandardComponent/mount" as="script" type="text/javascript"/><link rel="modulepreload" href="/components/StandardComponent/index.tsx" as="script" type="text/javascript"/><script type="text/javascript"></script><style>@import url(/components/Shared/styles.css);.foo{background-color:#55c456}body{background-color:#fbfbfb;background-image:url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDAgMTAwIj4KICA8IS0tIEJhc2UgdHJhbnNwYXJlbnQgYmFja2dyb3VuZCAtLT4KICA8cmVjdCB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgZmlsbD0idHJhbnNwYXJlbnQiLz4KICAKICA8IS0tIFN1YnRsZSBnZW9tZXRyaWMgcGF0dGVybiAtLT4KICA8ZyBmaWxsPSIjMzMzMzMzIiBmaWxsLW9wYWNpdHk9IjAuMDMiPgogICAgPCEtLSBTbWFsbCBjaXJjbGVzIC0tPgogICAgPGNpcmNsZSBjeD0iMTAiIGN5PSIxMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSIzMCIgY3k9IjEwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjUwIiBjeT0iMTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNzAiIGN5PSIxMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI5MCIgY3k9IjEwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMjAiIGN5PSIyMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI0MCIgY3k9IjIwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjYwIiBjeT0iMjAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iODAiIGN5PSIyMCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjEwIiBjeT0iMzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iMzAiIGN5PSIzMCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI1MCIgY3k9IjMwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjcwIiBjeT0iMzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iOTAiIGN5PSIzMCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjIwIiBjeT0iNDAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNDAiIGN5PSI0MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI2MCIgY3k9IjQwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjgwIiBjeT0iNDAiIHI9IjEuNSIvPgogICAgCiAgICA8Y2lyY2xlIGN4PSIxMCIgY3k9IjUwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjMwIiBjeT0iNTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNTAiIGN5PSI1MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI3MCIgY3k9IjUwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjkwIiBjeT0iNTAiIHI9IjEuNSIvPgogICAgCiAgICA8Y2lyY2xlIGN4PSIyMCIgY3k9IjYwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjQwIiBjeT0iNjAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNjAiIGN5PSI2MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI4MCIgY3k9IjYwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMTAiIGN5PSI3MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSIzMCIgY3k9IjcwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjUwIiBjeT0iNzAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iNzAiIGN5PSI3MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI5MCIgY3k9IjcwIiByPSIxLjUiLz4KICAgIAogICAgPGNpcmNsZSBjeD0iMjAiIGN5PSI4MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI0MCIgY3k9IjgwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjYwIiBjeT0iODAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iODAiIGN5PSI4MCIgcj0iMS41Ii8+CiAgICAKICAgIDxjaXJjbGUgY3g9IjEwIiBjeT0iOTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iMzAiIGN5PSI5MCIgcj0iMS41Ii8+CiAgICA8Y2lyY2xlIGN4PSI1MCIgY3k9IjkwIiByPSIxLjUiLz4KICAgIDxjaXJjbGUgY3g9IjcwIiBjeT0iOTAiIHI9IjEuNSIvPgogICAgPGNpcmNsZSBjeD0iOTAiIGN5PSI5MCIgcj0iMS41Ii8+CiAgICAKICAgIDwhLS0gVGhpbiBsaW5lcyAtLT4KICAgIDxsaW5lIHgxPSIwIiB5MT0iMjUiIHgyPSIxMDAiIHkyPSIyNSIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSIwIiB5MT0iNzUiIHgyPSIxMDAiIHkyPSI3NSIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSIyNSIgeTE9IjAiIHgyPSIyNSIgeTI9IjEwMCIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICAgIDxsaW5lIHgxPSI3NSIgeTE9IjAiIHgyPSI3NSIgeTI9IjEwMCIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utb3BhY2l0eT0iMC4wMiIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KICA8L2c+Cjwvc3ZnPg==");background-repeat:repeat;background-size:200px 200px;background-attachment:fixed}</style></head><body><main id="fullsoak-htmlshell-main"><div>StandardComponent=</div></main><script type="text/javascript">window.preloadedProps = {}</script><script defer type="module" src="/components/StandardComponent/mount"></script></body></html>'`;
